src/main/java/io/confluent/kafkarest/resources/ConsumersResource.java

```diff
@@ -55,23 +56,20 @@
 @Consumes({Versions.KAFKA_V1_JSON_BINARY, Versions.KAFKA_V1_JSON_AVRO, Versions.KAFKA_V1_JSON_JSON,
            Versions.KAFKA_V1_JSON, Versions.KAFKA_DEFAULT_JSON, Versions.JSON,
            Versions.GENERIC_REQUEST})
+@Singleton
 public class ConsumersResource {
 
-  private final Context ctx;
 
-  public ConsumersResource(Context ctx) {
-    this.ctx = ctx;
+  public ConsumersResource() {
   }
 
   @POST
   @Valid
   @Path("/{group}")
   @PerformanceMetric("consumer.create")
   public CreateConsumerInstanceResponse createGroup(
-      @javax.ws.rs.core.Context UriInfo uriInfo,
-      final @PathParam("group") String group,
-      @Valid ConsumerInstanceConfig config
-  ) {
+      @javax.ws.rs.core.Context UriInfo uriInfo, final @PathParam("group") String group,
+      @Valid ConsumerInstanceConfig config, @javax.ws.rs.core.Context Context ctx) {
```

Comment:
My first thought here is that it would seem more natural to have only things specific to the request come through as parameters. i.e. have the request context come through, not the rest proxy context. To make this work, there'd need to be something like a RestContextManager. Usage here, something like:

`RestContextManager.GetContext(requestContext).getConsumerManager()`

This is more long winded than what you've got but on the upside it's descriptive of what's going on. 

In `KafkaRestApplication.setupInjectedResources()`, a `RestContextManager` could be created that is either a default or if the security plugin is found, the custom one. The `RestResourceExtension` class would need a new method `getRestContextManagerInstance()`, or something.